utils/xbox.ixx

```diff
@@ -0,0 +1,125 @@
+module;
+#include <algorithm>
+#include <cstdint>
+#include <vector>
+
+export module utils.xbox;
+
+
+
+namespace gpsxre
+{
+
+export enum class XGD_Type : uint8_t
+{
+    UNKNOWN,
+    XGD1,
+    XGD2,
+    XGD3
+};
+
+export XGD_Type get_xgd_type(const std::vector<uint8_t> &ss)
```

Comment:
For this check we are checking LayerDescription, I will update this.

Reply:
I think it may be best to pass the SS as a `READ_DVD_STRUCTURE_LayerDescriptor` and then cast the media_specific portion to a struct.
The reason for this is to not duplicate the definition of the LayerDescriptor portion, and that there are 3 different structs for the media_specific portion depending on the XGD type.